src/drivers/mct/main/component_mod.F90

```diff
@@ -137,61 +135,40 @@ subroutine component_init_pre(comp, compid, cplcompid, cplallcompid, &
        allocate(comp(eci)%dom_cc)
        allocate(comp(eci)%gsmap_cc)
        allocate(comp(eci)%cdata_cc)
-       comp(eci)%cdata_cc%name     = 'cdata_'//ntype(1:1)//ntype(1:1)
-       comp(eci)%cdata_cc%ID       =  comp(eci)%compid
-       comp(eci)%cdata_cc%mpicom   =  comp(eci)%mpicom_compid
-       comp(eci)%cdata_cc%dom      => comp(eci)%dom_cc
-       comp(eci)%cdata_cc%gsmap    => comp(eci)%gsmap_cc
-       comp(eci)%cdata_cc%infodata => infodata
-
-       ! Does this component need to do post-data assimilation processing?
-       if (seq_timemgr_data_assimilation_active(ntype(1:3))) then
-          comp_resume(:) = 'TRUE'
-       else
-          comp_resume(:) = ''
-       end if
+       call seq_cdata_init(comp(eci)%cdata_cc, comp(eci)%compid,              &
+            'cdata_'//ntype(1:1)//ntype(1:1), comp(eci)%dom_cc,               &
+            comp(eci)%gsmap_cc, infodata, seq_timemgr_data_assimilation_active(ntype(1:3)))
```

Comment:
For some reason seq_cdata_init was not being used. I have resurrected it and added the new arguments.